### PR TITLE
Deprecate bap modal android

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/13798

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java [hunk 1]
```diff
@@ -372,6 +389,67 @@ public void finishNativeInitialization() {
                     calender.getTimeInMillis());
         }
         checkSetDefaultBrowserModal();
+        if (mBraveRewardsNativeWorker != null) {
+            mBraveRewardsNativeWorker.StartProcess();
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/app/BraveActivity.java:393]: Maybe we shouldn't start it where rewards are not available, x86 devices?

Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/app/BraveActivity.java:393]: Good point. updating it.

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by LaurenWags** [comment]
Verified using LG Nexus 5 with Android 5.1 running 1.22.22 Bravearm.apk


Verified test plan from https://github.com/brave/brave-browser/issues/13798

<details>
<summary>Scenario 1: Non-JP Regions</summary>

Checked with US/English

Confirmed able to claim staging promotion:

Example| Example | Example
----- | ----- | ----- 
US1 | US2 | US3

Confirmed BAP popup does not appear.
Confirmed BAP alert does not display on NTP when system date is set to 2021-03-06.
Confirmed wallet balance is not zero when system date is set to 2021-03-13:

Example| Example 
----- | ----- 
US4 | US5

</details>

<details>
<summary>Scenario 2: JP Region</summary>

Checked with Japan/Japanese (unable to check with Japan/English on my device). Note, issue encountered where Japanese translations are missing, logged with https://github.com/brave/brave-browser/issues/14117.

Open NTP with zero BAP balance. Confirmed no BAP popup.

Confirmed able to claim UGP grant, balance is now non-zero (30 BAP)

Example| Example | Example
----- | ----- | ----- 
 |  | 

Confirmed BAP popup displays when there is a non-zero BAP balance:

Portrait | Landscape
----- | -----
tooltip-portrait | tooltip-landscape

Confirmed tapping outside of the BAP popup closes the popup (there is no "x" to close).
Confirmed clicking on "Learn More" shows the BAP alert:

Portrait | Landscape
----- | -----
modal-portrait | modal-landscape

Confirmed able to close the BAP alert via "OK".
Confirmed opening a NTP did not show the BAP popup again.
Confirmed in 3 days the BAP popup shows on NTP again (advanced device clock per test plan).

Confirmed when computer date is 2021-03-06 and Brave is opened, the BAP alert is displayed and the BAP popup is not.
Confirmed able to close the BAP alert via "OK".
Confirmed opening a NTP did not show the BAP popup again if it had been dismissed.
Confirmed in 3 days the BAP popup shows on NTP again (advanced device clock per test plan).

Confirmed when computer date is 2021-03-13 and Brave is opened, the BAP popup and alert are not displayed.
Confirmed that the BAP balance is now zero on the panel and brave://rewards:

Example| Example
----- | -----
 | 

</details>

Encountered the following while testing Scenario 2

- BAP modal issues - https://github.com/brave/brave-browser/issues/14136
- Japanese translations missing for BAP popup and alert - https://github.com/brave/brave-browser/issues/14117

**Comment by srirambv** [comment]
Verification passed on OnePlus 6T with Android 10 running 1.22.24 x64 Nightly build

<details>
<summary><b>Scenario 1: Non-JP region</b></summary>

- Verified able to claim promotions 
- Verified no BAP modal is shown
- Verified no BAP alert is shown when device date is set to 3 Mar 2021
- Verified no BAP alert is shown when device date is set to 14 Mar 2021
- Verified wallet balance doesn't change or resets to zero when device date is set to 14 Mar 2021

image | image | image
-- | -- | --


</details>
<details>
<summary><b>Scenario 2: JP region</b></summary>

- Verified setting device locale to JP shows and clean install with 0 BAP balance doesn't show any BAP deprecation modal
- Verified able to claim grant and wallet balance updates
- Verified upgrading to 1.22.41 with a wallet balance shows the deprecation modal upon launch
- Verified modal shows up again when the device date is moved ahead by 3 days
- Verified clicking on learn more brings up a bigger modal with details 
- Verified tapping outside the modal closes it 
- Verified clicking only on OK closes the BAP info modal and not tapping outside the modal 
- Verified moving device date to 6 Mar shows only the BAP alert not the tooltip on Rewards
- Verified moving device date to 13 Mar doesn't show BAP alert or tooltip and wallet balance is reset to 0
- Verified when device date/time is set to Mar 12 11.59PM, keeping the browser active with a wallet balance is reset to 0 without having to restart the browser

image | image | image | image | 
-- | -- | -- | -- 


</details>

<b>

- Encountered https://github.com/brave/brave-browser/issues/14117
- Encountered https://github.com/brave/brave-browser/issues/14136#issuecomment-779224711
- Logged follow-up issue https://github.com/brave/brave-browser/issues/14158
</b>

**Pull request merged**